endpoint/domain_filter_test.go

```diff
@@ -668,3 +668,104 @@ func deserialize[T any](t *testing.T, serialized map[string]T) DomainFilter {
 
 	return deserialized
 }
+
+func TestDomainFilterMatchParent(t *testing.T) {
+	parentMatchTests := []domainFilterTest{
+		{
+			[]string{"a.example.com."},
```

Comment:
can you use the key-value style assignment please?
It's hard to read from the PR what all these members of `domainFilterTest` will be

Reply:
Sure. I'm going to refactor it.

Reply:
@szuecs This test was removed from the [upstream](https://github.com/kubernetes-sigs/external-dns/commit/7aa9d25067c0d356095cfd2dd0b334a9c7f20a4f#diff-ccbfca3f7dcafb10f67bd064fa6fdebdaba4fcc509aa293d3cf6279ebdfd8740L442) on August 24, and it remains the same without any change, I just recovered it. Do you think it must be refactored?

Reply:
I guess it would be better, but from my side not a strict requirement.